### PR TITLE
Output GitHub contexts

### LINKED ISSUE #5: Show worflow info, not only python info.
Klein has this extra info - https://github.com/twisted/klein/blob/4e6bb27536d7437ac9eecab76b9c00bbd9ee103c/.github/workflows/cicd.yml#L18

Once this is done, use it in Klein  https://github.com/twisted/klein/issues/458

### PR DESCRIPTION
**Pull request by altendky**
Closes https://github.com/twisted/python-info-action/issues/5

### CODE REVIEW

**Comment by altendky** [comment]
Kick CI after re-enabling (was disabled due to inactivity)

File: info.py [hunk 1]
```diff
@@ -28,45 +30,54 @@ def print(self, *args, **kwargs):
...
+            )
+            self.print(line)
+
+
 output = Output()
 
-output.heading("Python Details")
+environment = dict(os.environ)
+
+context_prefix = "_PYTHON_INFO_ACTION_CONTEXT_"
```
Comment (adiroiban) [inline info.py]: Maybe rename to uppercase and move it at the start of the file.

This looks like a configuration value for an external decency.

```suggestion
CONTEXT_PREFIX = "_PYTHON_INFO_ACTION_CONTEXT_"
```

File: info.py [hunk 2]
```diff
@@ -28,45 +30,54 @@ def print(self, *args, **kwargs):
         sys.stdout.write(chunk)
         self.chunks.extend(chunk)
 
-    def heading(self, name):
+    def heading(self, name, level):
```
Comment (adiroiban) [inline info.py:35]: Maybe add GitHub log group marker for each top level heading

```suggestion
    def heading(self, name, level):
        if level == 0:
            self.print('::endgroup::')
            self.print('::group::{}'.format(name))
```

**Review by adiroiban — APPROVED** [review]
Looks good. Thanks.

Looks like steps and job don't contain much info but maybe they are used to debug  complex workflow.

Maybe @wsanchez can take a look and see if anything is missing or blocking the usage of this action in klein.

btw, for twisted/twisted maybe we should use `v1` tag and update the `v1` with each release that is backward compatible... otherwise we will need a new PR or drive fix for each update of this action.

File: info.py (hunk 1 above)
Comment (altendky) [reply info.py]: I've read and written too many C macros and thus have a visceral reaction to `YELLING_HACKY_GARBAGE_AUAAUUGHHGHGH`...  but yes, that isn't what it is in Python and it is common.

File: info.py (hunk 2 above)
Comment (altendky) [reply info.py:35]: I'm tempted to exclude this from the file output.  The only problem I see is if people hope to capture the output the `cat` it later or somesuch and retain the groupings.  But, that doesn't seem particularly likely.

Is the `pip install something` bit actually meant to go here?  Or just part of an example from elsewhere?

**Comment by altendky** [comment]
Hmm, maybe the reST-like heading should be for the file only?  I dunno...

image
image

File: info.py (hunk 2 above)
Comment (adiroiban) [reply info.py:35]: Ups... sorry. Copy/paste error. I have updated the suggestion.
Feel free to exclude it :)

**Review by adiroiban — APPROVED** [review]
merge, merge, merge ! :) Thanks!

**Pull request merged**